python/ray/train/torch/train_loop_utils.py

```diff
@@ -365,6 +369,12 @@ def __init__(self, amp: bool = False):
         self.amp_is_enabled = amp
         self.scaler = GradScaler() if amp else None
         self._seed = None
+        # default is CUDADeviceManager
+        session = get_session()
+        if session and session.device_manager:
+            self.device_manager = session.device_manager
```

Comment:
do not store this global variable as an attribute of this object, just call the global getter each time.

Reply:
Done.